### PR TITLE
motion estimation restructuring 

### PR DESCRIPTION
**Pull request by NaderMahdi**
Restructure the full-pel motion search to perform 8 8x8 SAD in //.

Off by default and to be enabled after the intrinsics is added.

### CODE REVIEW

File: Source/Lib/Codec/EbMotionEstimation.c [hunk 1]
```diff
@@ -1059,15 +1108,1356 @@ static EB_EXTSADCALCULATION_TYPE ExtSadCalculation_funcPtrArray[ASM_TYPE_TOTAL]
...
+
+        sad_16x8[17] = p_sad8x8[34][search_index] + p_sad8x8[35][search_index];
+        if (sad_16x8[17] < p_best_sad16x8[17]) {
+            p_best_sad16x8[17] = sad_16x8[17];
+            x_mv = _MVXT(mv) + (int16_t)search_index * 4;
+            y_mv = _MVYT(mv);
+            p_best_mv16x8[17] = ((uint16_t)y_mv << 16) | ((uint16_t)x_mv);
+        }
+
+        sad_16x8[18] = p_sad8x8[36][search_index] + p_sad8x8[37][search_index];
```
Comment (mafonsoN) [inline Source/Lib/Codec/EbMotionEstimation.c]: There is a lot of repeated code here. Can this be done using a loop instead?

Comment (hassount) [reply Source/Lib/Codec/EbMotionEstimation.c]: I believe this par of the code is there on purpose as it makes the code easy to read / test particular sad locations / partitions.

**Comment by hassount** [comment]
@joelsole @andrey-norkin @kylophone @mafonsoN 
Can we move forward with this PR?

**Comment by joelsole** [comment]
I believe so. This was reviewed by Mariana a few days ago.

**Pull request merged**